__tests__/attachEventListeners.test.js

```diff
@@ -0,0 +1,89 @@
+import { JSDOM } from "jsdom";
+import { describe, expect, test, afterEach, vi, beforeAll } from "vitest";
+import { attachEventListeners, initializeElementsReferences } from "../src/script";
+import handleFormSubmit from '../src/handleFormSubmit';
+
+describe("attachEventListeners", () => {
+
+  global.chrome = {
+    storage: {
+      sync: {
+        set: () => { }
+      }
+    },
+    scripting: {
+      executeScript: () => { }
+    },
+    tabs: {
+      query: () => [{ id: 1 }]
+    }
+  };
+
+  const setupDOM = () => {
+    const dom = new JSDOM(`
+      <div id="container">
+        <input id="input" type="text" />
+        <button id="button"><span>Find</span></button>
+      </div>
+    `);
+    global.document = dom.window.document;
+    return { dom };
+  };
+
+  test('Should attach event listeners to input element', () => {
+    setupDOM();
```

Comment:
can we move this to beforeEach?

Reply:
I don't think its a good idea, setupDOM function is used to initialise elements and it returns them.
I'm trying to avoid declaring a global variables, if you want to use them I can switch that.

Reply:
we could nest this in describe and still move it to before each in this nested tests, I can imagine that this cases could grow and we would avoid writing the same lines again